### PR TITLE
Swap UX Cleanup

### LINKED ISSUE #226: Buggy swap error message
I'm on a pretty slow connection, so this might just be an issue that happens when you haven't loaded exchange rates. But putting a 1 into the left field (btc) prompted me with this:

screen shot 2017-09-23 at 1 30 15 pm

### LINKED ISSUE #383: Different ETH/BTC Ratio vs Production for Swap
As seen in the screenshot, the calculation of ETH/BTC ratio is off, possibly from a different API endpoint call than production site

2017-11-11-214505_1684x776_scrot

### PR DESCRIPTION
**Pull request by eddiewang**
Fixes #226 
Fixes #383 

Added a simple check to ensure that swap rates exist so we don't get a NaN error, which would be confusing to users.

Instead, the form is hidden and a spinner is shown until the bityRates are ready for the user.

### CODE REVIEW

**Comment by eddiewang** [comment]
@dternyak if we add the err messages to the input fields do we still want to display the notification bar at the bottom?

**Comment by dternyak** [comment]
Nope!

File: common/containers/Tabs/Swap/components/CurrencySwap.tsx [hunk 1]
```diff
@@ -77,31 +82,39 @@ export default class CurrencySwap extends Component<
...
+          console.log('Something went terribly wrong!');
+      }
+      const errString = `Min ${minAmount} ${originKind}. Max ${maxAmount} ${originKind}.`;
+
+      this.setState({
+        originErr: errString
+      });
     } else {
       this.setState({
+        originErr: ''
```
Comment (HenryNguyen5) [inline common/containers/Tabs/Swap/components/CurrencySwap.tsx]: Should combine these two setStates into one

File: common/containers/Tabs/Swap/components/CurrencySwap.tsx [hunk 2]
```diff
@@ -77,31 +82,39 @@ export default class CurrencySwap extends Component<
...
+          break;
+        case 'ETH':
+          minAmount = toFixedIfLarger(ETHMin, 3);
+          maxAmount = toFixedIfLarger(ETHMax, 3);
+          break;
+        case 'REP':
+          minAmount = toFixedIfLarger(REPMin, 3);
+          break;
+        default:
+          console.log('Something went terribly wrong!');
```
Comment (HenryNguyen5) [inline common/containers/Tabs/Swap/components/CurrencySwap.tsx]: Got a console log here

File: common/components/ui/DropdownShell.tsx [hunk 3]
```diff
@@ -45,7 +53,9 @@ export default class DropdownComponent extends Component<Props, State> {
 
     return (
       <span
-        className={`dropdown ${expanded ? 'open' : ''}`}
+        className={`dropdown ${expanded ? 'open' : ''} ${className
```
Comment (HenryNguyen5) [inline common/components/ui/DropdownShell.tsx]: Using `classnames` package here would clean it up a bit

File: common/components/ui/Dropdown.tsx [hunk 4]
```diff
@@ -3,6 +3,7 @@ import classnames from 'classnames';
 import DropdownShell from './DropdownShell';
 
 interface Props<T> {
+  className?: string;
```
Comment (wbobeirne) [inline common/components/ui/Dropdown.tsx]: Using `className` in custom props is kind of a faux paus, most components don't do it. If you want to alter the way it looks, you typically pass a more specific prop, like `color` or `size`. If you want to alter the way it's positioned or spaced, you typically wrap a `<div>` around the component and adjust the sizing or spacing of that instead.

Basically the problem is that if there's some CSS that the component relies on, i.e. negative margins or some positioning tricks, you might override it with your custom class. It makes reusable components a little hairier if they can get overridden like that.

File: common/containers/Tabs/Swap/components/CurrencySwap.tsx (hunk 2 above)
Comment (wbobeirne) [reply common/containers/Tabs/Swap/components/CurrencySwap.tsx]: I think that's intentional, but to make that clearer I would use `console.warn` and be a little more descriptive in the message.

File: common/components/WalletDecrypt/Keystore.tsx [hunk 5]
```diff
@@ -54,9 +54,9 @@ export default class KeystoreDecrypt extends Component {
             <div className={file.length && passReq ? '' : 'hidden'}>
               <p>{translate('ADD_Label_3')}</p>
               <input
-                className={`form-control ${
-                  password.length > 0 ? 'is-valid' : 'is-invalid'
-                }`}
+                className={`form-control ${password.length > 0
```
Comment (dternyak) [inline common/components/WalletDecrypt/Keystore.tsx]: only formatting

File: common/components/Header/index.tsx [hunk 6]
```diff
@@ -111,9 +111,7 @@ export default class Header extends Component<Props, {}> {
 
               <div className="Header-branding-right-dropdown">
                 <ColorDropdown
-                  ariaLabel={`change node. current node ${
-                    selectedNode.network
-                  } node by ${selectedNode.service}`}
+                  ariaLabel={`change node. current node ${selectedNode.network} node by ${selectedNode.service}`}
```
Comment (dternyak) [inline common/components/Header/index.tsx]: only formatting

File: common/components/ui/DropdownShell.tsx [hunk 7]
```diff
@@ -2,6 +2,7 @@ import React, { Component } from 'react';
 import classnames from 'classnames';
 
 interface Props {
+  className?: string;
```
Comment (dternyak) [inline common/components/ui/DropdownShell.tsx]: ?

File: common/containers/Tabs/SendTransaction/index.tsx [hunk 8]
```diff
@@ -316,14 +316,14 @@ export class SendTransaction extends React.Component<Props, State> {
...
                     {(offline || forceOffline) && (
-                      <div>
-                        <NonceField
-                          value={nonce}
-                          onChange={this.onNonceChange}
-                          placeholder={'0'}
-                        />
-                      </div>
-                    )}
+                        <div>
```
Comment (dternyak) [inline common/containers/Tabs/SendTransaction/index.tsx]: only formatting

File: common/derivation-checker.ts [hunk 9]
```diff
@@ -18,9 +18,7 @@ function promiseFromChildProcess(command): Promise<any> {
 }
 
 async function privToAddrViaDocker(privKeyWallet: IFullWallet) {
-  const command = `docker run -e key=${privKeyWallet.getPrivateKeyString()} ${
-    dockerImage
-  }:${dockerTag}`;
+  const command = `docker run -e key=${privKeyWallet.getPrivateKeyString()} ${dockerImage}:${dockerTag}`;
```
Comment (dternyak) [inline common/derivation-checker.ts]: only formatting

File: common/libs/wallet/non-deterministic/web3.ts [hunk 10]
```diff
@@ -71,9 +71,8 @@ export default class Web3Wallet implements IWallet {
         if (this.network !== networkName) {
           return reject(
             new Error(
-              `Expected MetaMask / Mist network to be ${
-                this.network
-              }, but got ${networkName}. ` +
+              `Expected MetaMask / Mist network to be ${this
```
Comment (dternyak) [inline common/libs/wallet/non-deterministic/web3.ts]: only formatting

File: spec/utils/formatters.spec.ts [hunk 11]
```diff
@@ -72,9 +72,7 @@ describe('formatNumber', () => {
 
   pairs.forEach(pair => {
     const digits = pair.digits;
-    it(`should convert ${pair.input.toString()} to ${pair.output} when using ${
-      digits
-    } digits`, () => {
+    it(`should convert ${pair.input.toString()} to ${pair.output} when using ${digits} digits`, () => {
```
Comment (dternyak) [inline spec/utils/formatters.spec.ts]: only formatting

**Pull request merged**